src/index.js

```diff
@@ -129,16 +129,28 @@ class LocationProvider extends React.Component {
 }
 
 ////////////////////////////////////////////////////////////////////////////////
-let ServerLocation = ({ req, children }) => {
-  const parsedUri = urlPgk.parse(req.url);
+let ServerLocation = ({ url, query, children }) => {
+  let searchIndex = url.indexOf("?");
+  let searchExists = searchIndex > -1;
+  let pathname;
+  let search = "";
+  let hash = "";
+
+  if (searchExists) {
+    pathname = url.substring(0, searchIndex);
+    search = url.substring(searchIndex);
+  } else {
+    pathname = url;
+  }
+
   return (
     <LocationContext.Provider
       value={{
         location: {
-          pathname: parsedUri.pathname,
-          search: parsedUri.search || "",
-          hash: parsedUri.hash || "",
-          query: req.query
+          pathname,
+          search,
+          hash,
+          query
```

Comment:
The only thing that broke while merging reach/router into our package was the automatic passing of the query in the LocationContext.Provider.

It's been fixed here and addtionally in the rasch-stack [here](https://github.com/rasch-dtc/rasch-stack/pull/3329)